pandas/io/formats/info.py

```diff
@@ -72,92 +83,147 @@ def _sizeof_fmt(num: Union[int, float], size_qualifier: str) -> str:
     return f"{num:3.1f}{size_qualifier} PB"
 
 
-class BaseInfo(metaclass=ABCMeta):
+class BaseInfo(ABC):
+    """Base class for DataFrameInfo and SeriesInfo.
+
+    Parameters
+    ----------
+    data : FrameOrSeries
+        Either dataframe or series.
+    memory_usage : bool or str, optional
+        If "deep", introspect the data deeply by interrogating object dtypes
+        for system-level memory consumption, and include it in the returned
+        values.
+    """
+
     def __init__(
         self,
         data: FrameOrSeries,
-        verbose: Optional[bool] = None,
-        buf: Optional[IO[str]] = None,
-        max_cols: Optional[int] = None,
         memory_usage: Optional[Union[bool, str]] = None,
-        null_counts: Optional[bool] = None,
     ):
-        if buf is None:  # pragma: no cover
-            buf = sys.stdout
+        self.data = data
+        self.memory_usage = self._initialize_memory_usage(memory_usage)
+
+    @staticmethod
+    def _initialize_memory_usage(
+        memory_usage: Optional[Union[bool, str]] = None,
+    ) -> Union[bool, str]:
         if memory_usage is None:
             memory_usage = get_option("display.memory_usage")
+        return memory_usage
 
-        self.data = data
-        self.verbose = verbose
-        self.buf = buf
-        self.max_cols = max_cols
-        self.memory_usage = memory_usage
-        self.null_counts = null_counts
+    @property
+    @abstractmethod
+    def ids(self) -> Index:
+        """Column names or index names."""
+
+    @property
+    @abstractmethod
+    def dtype_counts(self) -> Mapping[str, int]:
+        """Mapping dtype - number of counts."""
 
+    @property
     @abstractmethod
-    def _get_mem_usage(self, deep: bool) -> int:
+    def non_null_counts(self) -> Sequence[int]:
+        """Sequence of non-null counts for all columns or column (if series)."""
+
+    @property
+    @abstractmethod
+    def dtypes(self) -> "Series":
+        """Dtypes.
+
+        Returns
+        -------
+        dtypes : Series
+            Dtype of each of the DataFrame's columns.
         """
-        Get memory usage in bytes.
+        return self.data.dtypes
 
-        Parameters
-        ----------
-        deep : bool
-            If True, introspect the data deeply by interrogating object dtypes
-            for system-level memory consumption, and include it in the returned
-            values.
+    @property
+    def mem_usage(self) -> int:
```

Comment:
would rather not abbeviate this method (so maybe have to use _memory_usage for the storage

Reply:
I renamed it to ``memory_usage_bytes``.